processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -73,9 +77,14 @@
   private final Supplier<DruidFloatPredicate> floatPredicateSupplier;
   private final Supplier<DruidDoublePredicate> doublePredicateSupplier;
 
+  @JsonIgnore
+  private byte[] cacheKey;
+
   @JsonCreator
   public InDimFilter(
       @JsonProperty("dimension") String dimension,
+      // This 'values' collection instance can be reused if possible to avoid copying a big collection.
+      // Callers should _not_ modify the collection after it is passed to this constructor.
       @JsonProperty("values") Collection<String> values,
```

Comment:
Can we pass in a `Set<String>` instead of `Collection<String>` so jackson can do the de-dupe for us?

Reply:
To be honest, I was lazy and haven't checked how many callers call this constructor before. It seems small enough, I guess I can.

Reply:
Done.